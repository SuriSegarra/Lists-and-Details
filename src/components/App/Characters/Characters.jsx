import React from 'react';
import PropTypes from 'prop-types';

const Character =({ name, image }) => (
  <figure>
    <img src={image} alt=''/>
    <figcaption>
      <h1>{name}</h1>
    </figcaption>
  </figure>
);

Character.propTypes = {
  name: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired
};

export default Character;
