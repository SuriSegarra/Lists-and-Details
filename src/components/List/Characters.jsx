
import React from 'react';
import Character from '../Detail/Character';
import PropTypes from 'prop-types';

const Characters = ({ characters }) => {
  const charactersList = characters.map(character => (
    <li key={character.id}>
      <Character {...characters}/>
    </li>
  ));
  return (
    <ul>
      {charactersList}
    </ul>
  );
};

Characters.propTypes = {
  characters: PropTypes.arrayOf(PropTypes.shape({

    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    image: PropTypes.string.isRequired
  })).isRequired

};

export default Characters;
