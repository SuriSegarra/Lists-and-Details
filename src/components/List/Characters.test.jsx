import React from 'react';
import { shallow } from 'enzyme';
import Characters from './Characters';

const characters = [
  {
   
    name: 'Ang',
    image: 'url'
  }
];

describe('Characters', () => {
  it('renders Characters', () => {
    const wrapper = shallow(<Characters characters={characters} />);
    expect(wrapper).toMatchSnapshot();
  });
});
