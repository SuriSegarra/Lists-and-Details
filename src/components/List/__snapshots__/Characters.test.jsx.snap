// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character renders Characters 1`] = `
<ul
  className="Characters"
>
  <Link
    to="/character/undefined"
  >
    <li>
      <Character
        id={1}
        image="url"
        name="Ang"
      />
    </li>
  </Link>
</ul>
`;

exports[`Characters renders Characters 1`] = `
<ul
  className="Characters"
>
  <Link
    to="/character/undefined"
  >
    <li>
      <Character
        image="url"
        name="Ang"
      />
    </li>
  </Link>
</ul>
`;
